config/policies.go

```diff
@@ -0,0 +1,156 @@
+package config
+
+import (
+	"golang.ngrok.com/ngrok/internal/pb"
+)
+
+type policies struct {
+	Inbound  []*policy
+	Outbound []*policy
+}
+type policy struct {
+	Name        string
+	Expressions []string
+	Actions     []*action
+}
+type action struct {
+	Type   string
+	Config string
+}
+
+type PoliciesOption func(*policies)
+type PolicyOption func(*policy)
+type ActionOption func(*action)
+
+// WithPolicies creates a new set of policies with the provided options
+func WithPolicies(opts ...PoliciesOption) interface {
+	HTTPEndpointOption
+	TLSEndpointOption
+	TCPEndpointOption
+} {
+	p := &policies{}
+
+	for _, opt := range opts {
+		opt(p)
+	}
+
+	return p
+}
+
+// WithInboundPolicy adds the provided policy to be applied on inbound connections on an ngrok edge.
+// The order in which policies are added is respected at runtime.
+func WithInboundPolicy(opts ...PolicyOption) PoliciesOption {
+	return func(p *policies) {
+		inP := &policy{}
+		for _, opt := range opts {
+			opt(inP)
+		}
+
+		p.Inbound = append(p.Inbound, inP)
+	}
+}
+
+// WithOutboundPolicy adds the provided policy to be applied on outbound connections on an ngrok edge.
+// The order in which policies are added is respected at runtime.
+func WithOutboundPolicy(opts ...PolicyOption) PoliciesOption {
+	return func(p *policies) {
+		outP := &policy{}
+		for _, opt := range opts {
+			opt(outP)
+		}
+		p.Outbound = append(p.Outbound, outP)
+	}
+}
+
+// WithName sets the provided name on this policy
+func WithName(name string) PolicyOption {
```

Comment:
Within a generic `config` package, options like this are themselves far too generic. Let's insert `Policy` between the `With` and *thing* for all of the `PolicyOption`s, e.g. `WithPolicyName`